### DISCUSSION TITLE
HAA SonOff SV disconnecting WIFI after 1 or 2 days

### DISCUSSION THREAD

**Discussion by Sandree80** [RavenSystem/esp-homekit-devices#1685] [HAA Home App]
Hello,

Please help, I have SonOff SVs with the newest HAA firmware and since I switched to ASUS RT-AX92U & RT-AX55 with AI Mesh after 1 or 2 days my devices are disconnected. After rebooting HAA devices everything goes back to normal for 1-2 days...
I really tried everything:
- firmware upgrade on routers (in every month)
- fw upgrade SonOff HAA (now those are HAA v9.5.1). 
- The router's syslog are collected but I don't see any issues
- I tried all the wifi modes normal / active roaming / passive / roaming / force bssid but non of them could solve my issue
- Optimize channel on routers...
- HAA devices were stick to one router to close out AI mesh related issue
now I am wondering how to collect logs from HAA to see what is happening there, or any good idea, tips are highly appreciated.

Thank you in advance,
Sándor

**Comment by mariusvasss** [comment]
Maybe 🤔 the router standard, try on 2.4 ghz just 'n' and not auto. I had a homepod issue with wifi because of that. 
Routers switch between b, g, n, ac, ax.... 
This can be a fix.

**Comment by Daveblanche** [comment]
I moved to unifi because of issues, but then found some information sources like https://homekit.blog/forum/how-i-completely-fixed-not-responding/  And  https://www.reddit.com/r/HomeKit/comments/b72vii/how_i_completely_fixed_not_responding/
And
https://www.reddit.com/r/HomeKit/comments/n4rgft/asus_routers_and_homekit_a_solution_not_another/

Give it a go. They might have saved me a small fortune…

**Comment by Sandree80** [comment]
Dear Mariusvasss & Daveblanche,

Thank you your fast feedback. 
I don't want to do multiple settings at the same time so I did the followings as a first step:
- Smart Connect is turned off (based on this recommendation by Asus: https://www.asus.com/support/FAQ/1042475/)
- On 2.4Ghz the Wireless mode was changed from "auto" to "N only" 

Result: Promising. After switching  to "N only" all disconnected HAA devices came back without reboot that was never happened before.

Next steps will be what Daveblanche suggested:
- https://homekit.blog/forum/how-i-completely-fixed-not-responding/ 
- https://www.reddit.com/r/HomeKit/comments/b72vii/how_i_completely_fixed_not_responding/
- https://www.reddit.com/r/HomeKit/comments/n4rgft/asus_routers_and_homekit_a_solution_not_another/
I have reviewed the above link and most of the paramters are already set.

I will keep you updated and thank you the assistance,
Sándor

**Reply by Sandree80** [reply]
Hi Guys,

I am back again to share the results: Unfortunately one of my HAA device interrupted again so I made the following changes based on Daveblanche's recommendations:

Wireless / Professional / Explicit Beamforming disable
Wireless / Professional / Universal Beamforming disable
Lan / IPTV / Enable multicast routing (IGMP Proxy): Enable

All others were already set or not available. 

If this does not help I don't know how to continue...

Best Regards,
Sándor

**Comment by Sandree80** [comment]
Hello Guys again,

The last three settings (2 beamforming and IPTV) did not help it is worst then ever: previously HAA devices could stay on the WiFi ~2 days now it was only a few hours.

What is also interesting, when HAA devices are note ping-able, but can send messages if door is opening or closing...

Any idea how to trace what is going inside HAA?

Thank you in advance,
Sándor

**Comment by i3laze** [comment]
Stay with Auto settings. Set fixed channel width: **20MHz** at 2.4GHz (not Auto, 40 or 20/40).

Sorry, can't suggest anything else without a radio scan) 
Is this an urban area with dozens of APs around or a stand-alone country house?

How about you bother Asus tech support on that issue? Provide them some logs from access point e.t.c..
Got similar hardware Xiaomi AX3600+AX1800 in Mesh with dozens of HAA devices without issues.

**Reply by Sandree80** [reply]
Hello i3laze,

Thank you for the tips in fact all recommended values are already set 20Mhz on 2,4Ghz Control Channel 6. I reduced the WPA2/WPA3 to WPA/WPA2. All HAA devices have proper signal strength and it seems the issue is idependent from WiFi signal.

On the weekend I was focusing on the logging: now all HAA devices and routers are sending logs to a central logserver. (routers have verbose logging)

At first sight from the logs there were two possible reasons:
- mDNS: I tried to switch off on HAA side but it was not possible so I made an attempt with minimal value 30s and with maximum 14976s now I kept the max value and it seems to be more reliable (with the minimal value after one hour HAA became unavailable)
- WPA: I saw some WPA realted issue on router: "_eth5: STA {MAC addr} WPA: group key handshake failed (RSN) after 4 tries_" so that is why I turned off WPA3 & disabled Protected Management Frames.

On this weekend there were several attempts which made the complete network unreliable now I stabilized and wait for the next occurrence with logging.

Best Regards,
Sándor

**Comment by RavenSystem** [comment]
FYI: mDNS is totally necessary for HomeKit and can not be disable.

I have 2 Asus RT-AX88U, but I know that this configuration works with other models, like AX86U, AC86, AC68, AC5300... This is my personal configuration. Never has disconnections. I use last version of firmware Asus-Merlin:

<img width="895" alt="Captura de pantalla 2022-01-18 a las 0 08 47" src="https://user-images.githubusercontent.com/8655274/149846235-d29ceb69-c341-49ea-afc9-8341e3bd3b37.png">
<img width="845" alt="Captura de pantalla 2022-01-17 a las 23 57 32" src="https://user-images.githubusercontent.com/8655274/149845753-b214f1bb-aa71-48a1-abbf-0273c2d8f7f6.png">
<img width="866" alt="Captura de pantalla 2022-01-17 a las 23 57 40" src="https://user-images.githubusercontent.com/8655274/149845757-37f8cb52-ec18-45fa-adca-535e7a87e71b.png">
<img width="942" alt="Captura de pantalla 2022-01-17 a las 23 57 51" src="https://user-images.githubusercontent.com/8655274/149845762-c43c7db1-0fd1-4dbb-9ce6-b6e8f3835225.png">

**Reply by Sandree80** [reply]
Hello,

Thank you for sharing this info but I have to say you have a luck with RT-AX88U, 'cause my  RT-AX92U which is not supported by Merlin :(

Best Regards,
Sándor

**Reply by RavenSystem** [reply]
Try configuring your router with same options.

**Comment by ghira89** [comment]
![image](https://user-images.githubusercontent.com/90135786/150343822-010f8b19-b4e1-493b-8fe1-5a4495860dce.png)

**Reply by ghira89** [reply]
Fritz box router 
Haa 9.5.1 
1 day to loss Wi-Fi

**Reply by ghira89** [reply]
X time to not used lamp Wi-Fi down.. 
but lamp Is in state on Wi-Fi Is ok not down x time..

**Comment by Sandree80** [comment]
Hi Guys Again,

Now I have to park the investigation due to other projects .But the problem is still exists.
I'll come within 2 weeks I don't want to give it up. ;)

Cheers,
S.

**Comment by Sandree80** [comment]
Hi Guys,

Finally I found the issue but unfortunately it is not on configuration side. As I mentioned I have an AI mesh Asus environment where the primary router is the better/stronger RT-AX92U and as a secondary a cheaper RT-AX55 is inuse. I always used the RT-AX92U 'cause I thought this has better signal strengh and much more reliable. Since I stuck all SonOff HAA to RTAX55 everything is smooth. That was my last chance :).

So now I have a new challenge to enforce the warranity on AX92U becuase from other aspects it's working smoothly, most probably just 2,4GHz part has an issue wit longterm permanent connections, but it is almost impossible to prove it. I've already created a ticket to Asus support since a month but there is no answer which is disappointing...

Anyway I am happy that all my HAA devices are healthy again and really appreciate your tips & support!

Bye,
Sándor
